index.rst

```diff
@@ -6,24 +6,474 @@
+
+::
+
+  plot_filter_colors = {'u': '#56b4e9', 'g': '#008060', 'r': '#ff4000', 'i': '#850000', 'z': '#6600cc', 'y': '#000000'}
+
+
+Image orientation
+^^^^^^^^^^^^^^^^^
+
+If using a WCS, display east left, north up.
+If only using pixels, (0,0) should be lower left, which is the default for awfDisplay.
+
+When using other plotting packages, transformations might be needed in order to match the afwDisplay default.
+See the LSST Science Pipelines documentation about `Image Indexing <https://pipelines.lsst.io/modules/lsst.afw.image/indexing-conventions.html>`_.
+
+Since use of "extent" is necessry for displaying a WCS overlay for deepCoaads, use it all the time:
+
+::
+
+  deepCoadd = butler.get('deepCoadd', dataId=dataId)
+  deepCoadd_bbox = butler.get('deepCoadd_calexp.bbox', dataId=dataId)
+  deepCoadd_wcs = butler.get('deepCoadd_calexp.wcs', dataId=dataId)
+  deepCoadd_WCSfMd = WCS(deepCoadd_wcs.getFitsMetadata())
+  deepCoadd_extent = (deepCoadd_bbox.beginX, deepCoadd_bbox.endX, deepCoadd_bbox.beginY, deepCoadd_bbox.endY)
+  plt.subplot(projection=deepCoadd_WCSfMd)
+  plt.imshow(deepCoadd.image.array, cmap='gray', vmin=0, vmax=2, extent=deepCoadd_extent, origin='lower')
+
+
+Clearing memory
+---------------
+
+These are optional methods for keeping memory use manageable in notebooks which are, e.g., demonstrating
+data visualization techniques with big datasets.
+
+To reduce the memory footprint of a notebook, remove figures once they're no longer needed.
+See the ``remove_figure`` function defined in the DP0 notebook `03_Image_Display_and_Manipulation.ipynb <https://github.com/rubin-dp0/tutorial-notebooks/blob/main/03a_Image_Display_and_Manipulation.ipynb>`__.
+
+*A method for clearing memory of, for example, large arrays that are not going to be used further on in the notebook is in development.*
+
+
+Assert statements
+-----------------
+
+Where essential, or where a very specific value is expected, the ``assert`` command can be used to demonstrate to users that a condition is true.
+
+For example, ``assert`` statements can be used to confirm that service objects like TAP are not ``None`` or ``null`` before moving on and using that instance,
+or to check that values meet expectations (e.g., total rows returned from a query).
+
+However, take care not to use ``assert`` statements when, e.g., querying dynamic (prompt) datasets.
+Consider more pedagogical alternatives when possible (e.g., printing schema columns would also fail if the TAP service was not instantiated).
+
+
+Known warnings
+--------------
+
+If a code cell produces a warning which is known, the preferred method to ignore warnings is to add a markdown cell,
+*before* the code cell which produces the warning, to tell the user it is OK to ignore.
+
+*The appropriate use-cases for ignoring categories of warnings (below) is still under consideration.*
+This is not preferred because ignore categories of warnings can allow real issues to go unnoticed.
```

Comment:
If this is not preferred behavior, there should be a more visual warning. Or we should reconsider placing it here, as we don't want to encourage bad habits.

Reply:
Added into the note